### PR TITLE
Update Valencia Cons<-->Prim

### PR DESCRIPTION
**Pull request by carmaza**
## Proposed changes

Update Valencia functions computing conservatives <--> primitives so Valencia can be evolved with the current infrastructure for evolution systems. 

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.cpp [hunk 1]
```diff
@@ -110,11 +111,16 @@ void PrimitiveFromConservative<ThermodynamicDim, Dim>::apply(
...
-                          10.0 * std::numeric_limits<double>::epsilon(),
-                          10.0 * std::numeric_limits<double>::epsilon(), 100);
+  DataVector z;
+  try {
+    // NOLINTNEXTLINE(clang-analyzer-core)
+    z = RootFinder::toms748(f_of_z, lower_bound, upper_bound,
+                            10.0 * std::numeric_limits<double>::epsilon(),
+                            10.0 * std::numeric_limits<double>::epsilon(), 100);
+  } catch (std::exception& exception) {
+    ERROR("Failed to find z with TOMS748 root finder. Got exception message:\n"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.cpp]: Not a necessary change, just a suggestion I'd like your thoughts on :) 

How about `Failed to find the intermediate variable z with TOMS748 root finder while computing the primitive variables from the conserved variables. Got exception...`?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.cpp (hunk 1 above)
Comment (carmaza) [reply src/Evolution/Systems/RelativisticEuler/Valencia/PrimitiveFromConservative.cpp]: I incorporated your suggestion. I squashed the change.

**Review by nilsdeppe — APPROVED** [review]
:+1: LGTM

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
ignoring travis timeouts

**Pull request merged**